File: ios/OTPPoc/OTPPoc/Models/DeviceModels.swift
import Foundation

struct AttestationChallengeResponse: Decodable {
    let challenge: String
    let challenge_id: String
}

struct AttestationVerifyRequest: Encodable {
    let challenge_id: String
    let device_id: String
    let challenge: String
    let signed_response: String
    let public_key: String
    let app_id: String
}

struct AttestationVerifyResponse: Decodable {
    let attestation_jwt: String
    let expires_in: Int
    let note: String?
}

File: android/app/src/main/java/com/otppoc/android/models/DeviceModels.kt
package com.otppoc.android.models

data class AttestationChallengeResponse(
    val challenge: String,
    val challenge_id: String
)

data class AttestationVerifyRequest(
    val challenge_id: String,
    val device_id: String,
    val challenge: String,
    val signed_response: String,
    val public_key: String,
    val app_id: String
)

data class AttestationVerifyResponse(
    val attestation_jwt: String,
    val expires_in: Int,
    val note: String?
)
